File: devnhwa/2024_02_27/Lv.0/길이에 따른 연산.java

/*
https://school.programmers.co.kr/learn/courses/30/lessons/181879

문제 설명
정수가 담긴 리스트 num_list가 주어질 때, 리스트의 길이가 11 이상이면 리스트에 있는 모든 원소의 합을 10 이하이면 모든 원소의 곱을 return하도록 solution 함수를 완성해주세요.

제한사항
2 ≤ num_list의 길이 ≤ 20
1 ≤ num_list의 원소 ≤ 9
num_list의 원소를 모두 곱했을 때 2,147,483,647를 넘는 입력은 주어지지 않습니다.
입출력 예
num_list	result
[3, 4, 5, 2, 5, 4, 6, 7, 3, 7, 2, 2, 1]	51
[2, 3, 4, 5]	120
입출력 예 설명
입출력 예 #1

리스트의 길이가 13이므로 모든 원소의 합인 51을 return합니다.
입출력 예 #2

리스트의 길이가 4이므로 모든 원소의 곱인 120을 return합니다.

*/
class Solution {
    public int solution(int[] num_list) {
        int answer = 0;

        if(num_list.length < 11) answer = 1;
        for(int num : num_list){
            if(num_list.length >= 11) answer = answer + num;
            else answer = answer * num;
        }
        return answer;
    }
}

File: decembaek/2024_03/2024_03_13/길이에_따른_연산.py
"""
https://school.programmers.co.kr/learn/courses/30/lessons/181879
프로그래머스 Lv.0

제목 : 길이에 따른 연산

"""

# decembaek 풀이
import math


def solution(num_list):
    answer = 1
    return sum(num_list) if len(num_list) >= 11 else math.prod(num_list)


# 다른사람
def solution(num_list):
    if len(num_list) >= 11:
        return eval("+".join(list(map(str, num_list))))
    else:
        return eval("*".join(list(map(str, num_list))))
